caddyhttp/proxy/upstream.go

```diff
@@ -138,6 +138,9 @@ func (u *staticUpstream) NewHost(host string) (*UpstreamHost, error) {
 				if uh.Unhealthy {
 					return true
 				}
+				if uh.Dynamic {
+					return false
+				}
```

Comment:
yeah, since caddy can't see the difference between 2 different backends with this unfortunately.